ykrt/src/compile/jitc_yk/codegen/x64/mod.rs

```diff
@@ -2916,25 +2928,45 @@ impl CompiledTrace for X64CompiledTrace {
                 libc::PROT_WRITE | libc::PROT_EXEC,
             )
         };
+        // Check that the target address is 8-byte aligned. This is required so we can patch
+        // this atomically.
+        let patch_addr = patch_addr as *mut u64;
+        debug_assert!(patch_addr.addr() % 8 == 0);
         if res == 0 {
             // Calculate the relative offset from the patch address to the address of the
             // side-trace. We need to add the size of the jump instruction to the patch address.
-            // FIXME: Could we end up with an offset bigger than 32 bits?
-            let offset = i32::try_from(staddr as i64 - (patch_addr as i64 + JMP_32_SIZE)).unwrap();
-
-            // Check that the target address is 8-byte aligned. This is required so we can patch
-            // this atomically.
-            let patch_addr = patch_addr as *mut u64;
-            debug_assert!(patch_addr.addr() % 8 == 0);
-
-            // Generate the JMP instruction.
-            let mut bytes: [u8; 8] = [NOP_OPCODE; 8]; // Fill with NOPs
-            bytes[0] = JMP_OPCODE;
-            bytes[1..5].copy_from_slice(&offset.to_ne_bytes());
-            let pval = u64::from_ne_bytes(bytes);
-            unsafe {
-                // Now write the new instruction in a single atomic operation.
-                *patch_addr = pval;
+            let offset = i32::try_from(staddr as i64 - (patch_addr as i64 + JMP_32_SIZE));
+            if let Ok(offset) = offset {
+                // Emit a jump to a 32-bit offset.
+                let mut bytes: [u8; 8] = [NOP_OPCODE; 8]; // Fill with NOPs
+                bytes[0] = JMP_OPCODE;
+                bytes[1..5].copy_from_slice(&offset.to_ne_bytes());
+                let pval = u64::from_ne_bytes(bytes);
+                unsafe {
+                    // Now write the new instruction in a single atomic operation.
+                    *patch_addr = pval;
```

Comment:
This isn't an atomic operation.

Reply:
Will look into `AtomicPtr` as discussed offline.

Reply:
Oh, I've forgotten about this too.

Reply:
[5ffe5fb](https://github.com/ykjit/yk/pull/1591/commits/5ffe5fbf8bae53e974f65f16d96cdfcec7198231)